raiden/network/rpc/client.py

```diff
@@ -631,7 +631,10 @@ def __init__(self, jsonrpc_client, registry_address, startgas=GAS_LIMIT,
 
     def manager_address_by_token(self, token_address):
         """ Return the channel manager address for the given token. """
-        return self.proxy.channelManagerByToken.call(token_address)
+        manager_address_encoded = self.proxy.channelManagerByToken.call(token_address)
+        if not manager_address_encoded or manager_address_encoded == '0x':
+            manager_address_encoded = self.add_token(token_address)
```

Comment:
This is wrong IMO, it's changing the `manager_address_by_token` contact, why should a function that get a contract that does not exist register it?